test/e2e-go/e2e_go_suite_test.go

```diff
@@ -0,0 +1,174 @@
+	RunSpecs(t, "E2EGo Suite")
+}
+
+var (
+	tc testutils.TestContext
+	// isPrometheusManagedBySuite is true when the suite tests is installing/uninstalling the Prometheus
+	isPrometheusManagedBySuite = true
+	// isOLMManagedBySuite is true when the suite tests is installing/uninstalling the OLM
+	isOLMManagedBySuite = true
+	// kubectx stores the k8s context from where the tests are running
+	kubectx string
+	// projectName is the name of the test project
+	projectName string
+)
+
+// BeforeSuite run before any specs are run to perform the required actions for all e2e Go tests.
+var _ = BeforeSuite(func(done Done) {
+	var err error
+
+	By("creating a new test context")
+	tc, err = testutils.NewTestContext("GO111MODULE=on")
+	Expect(err).NotTo(HaveOccurred())
+	Expect(tc.Prepare()).To(Succeed())
+	projectName = filepath.Base(tc.Dir)
+
+	By("checking the cluster type")
+	kubectx, err = tc.Kubectl.Command("config", "current-context")
+	Expect(err).Should(Succeed())
+
+	By("checking API resources applied on Cluster")
+	output, err := tc.Kubectl.Command("api-resources")
+	Expect(err).NotTo(HaveOccurred())
+	if strings.Contains(output, "servicemonitors") {
+		isPrometheusManagedBySuite = false
+	}
+	if strings.Contains(output, "clusterserviceversions") {
+		isOLMManagedBySuite = false
+	}
+
+	if isPrometheusManagedBySuite {
+		By("installing Prometheus")
+		Expect(tc.InstallPrometheusOperManager()).To(Succeed())
+
+		By("ensuring provisioned Prometheus Manager Service")
+		Eventually(func() error {
+			_, err := tc.Kubectl.Get(
+				false,
+				"Service", "prometheus-operator")
+			return err
+		}, 3*time.Minute, time.Second).Should(Succeed())
+	}
+
+	if isOLMManagedBySuite {
+		By("installing OLM")
+		Expect(tc.InstallOLM()).To(Succeed())
+	}
+
+	By("initializing a project")
+	projectName := filepath.Base(tc.Dir)
+	err = tc.Init(
+		"--project-version", "3-alpha",
+		"--repo", path.Join("github.com", "example", projectName),
+		"--domain", tc.Domain,
+		"--fetch-deps=false")
+	Expect(err).Should(Succeed())
+
+	// Todo: add to the init scaffold
+	By("by adding scorecard custom patch file")
+	err = tc.AddScorecardCustomPatchFile()
+	Expect(err).NotTo(HaveOccurred())
+
+	By("creating an API definition")
+	err = tc.CreateAPI(
+		"--group", tc.Group,
+		"--version", tc.Version,
+		"--kind", tc.Kind,
+		"--namespaced",
+		"--resource",
+		"--controller",
+		"--make=false")
+	Expect(err).Should(Succeed())
+
+	By("implementing the API")
+	Expect(kbtestutils.InsertCode(
+		filepath.Join(tc.Dir, "api", tc.Version, fmt.Sprintf("%s_types.go", strings.ToLower(tc.Kind))),
+		fmt.Sprintf(`type %sSpec struct {
+`, tc.Kind),
+		`	// +optional
+	Count int `+"`"+`json:"count,omitempty"`+"`"+`
+`)).Should(Succeed())
+
+	// todo(camilamacedo86): add sdk as lib to the project
+	// todo(camilamacedo86): replace watches in the controllers to use the InstrumentedEnqueueRequestForObject
+	// more info: https://github.com/operator-framework/operator-sdk/pull/3436/files
```

Comment:
IMO, this would add unnecessary complexity to the Go e2e for very limited value. We already test `InstrumentedEnqueueRequestForObject` in unit tests in `operator-lib` and in e2e tests for Helm and Ansible operators.

These e2e tests are meant to ensure that the scaffolding, packaging, and execution of Go projects works. Testing `InstrumentedEnqueueRequestForObject` here would be similar to testing controller-runtime's `EnqueueRequestsFromMapFunc` handler here. Both handlers are unrelated to this test.

Reply:
Agree.